controllers/front/validate.php

```diff
@@ -365,29 +370,29 @@ private function handleException(Exception $exception)
             $this->generateNewCart();
         }
 
-        header('HTTP/1.0 500 Internal Server Error');
-
-        echo json_encode([
-            'status' => false,
-            'httpCode' => 500,
-            'body' => [
-                'error' => [
-                    'message' => $exceptionMessageForCustomer,
+        $this->exitWithCustomStatus(
+            [
+                'status' => false,
+                'httpCode' => 500,
+                'body' => [
+                    'error' => [
+                        'message' => $exceptionMessageForCustomer,
+                    ],
                 ],
+                'exceptionCode' => $exception->getCode(),
+                'exceptionMessage' => $exception->getMessage(),
             ],
-            'exceptionCode' => $exception->getCode(),
-            'exceptionMessage' => $exception->getMessage(),
-        ]);
-        exit;
+            500,
+            'Internal Server Error');
```

Comment:
I see that you kept the previous behavior, but having two time the HTTP status + managing the related HTTP code message by yourselves will lead to errors sooner or later.

You may for instance:
* Send HTTP 200 with `Internal Server Error`
* Send the key `httpCode` as 500 but with the second parameter as 404.

I would recommend to stick to only one parameter, i.e having the httpCode in your array, and leave the management of the headers inside your parent class.

If you don't have custom HTTP response code, you could use [`http_response_code()`](https://www.php.net/manual/en/function.http-response-code.php)

Reply:
You're right 👍